.github/workflows/codeql.yml

```diff
@@ -0,0 +1,66 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ 'master' ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ 'master' ]
+  schedule:
+    - cron: '37 8 * * 0'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
```

Comment:
NB doesn't use swift anywhere, so this can be likely simplified and hardcoded to `ubuntu-latest`

Reply:
To be pedantic, we use Swift in at least two files for generating macOS launchers, but depending on current dev@ discussion they might end up being externalized anyway.